ui/js/services/jobfilters.js

```diff
@@ -407,15 +407,25 @@ treeherder.factory('thJobFilters', [
          */
         function getFieldFiltersArray() {
             const fieldFilters = [];
+            const clopt = thClassificationTypes.classificationOptions;
 
             _.each($location.search(), function (values, fieldName) {
                 if (_isFieldFilter(fieldName)) {
-                    const valArr = _toArray(values);
+                    var valArr = _toArray(values);
                     _.each(valArr, function (val) {
+                        var text = '';
+                        if (_withoutPrefix(fieldName) === 'failure_classification_id') {
+                            for (var i = 0; i < clopt.length; i++) {
+                                if (clopt[i]['id'].toString() === val) {
```

Comment:
when referring to field names, it's preferred to use: ``thisClopt.id`` rather than ``['id']``